### PR TITLE
feat(pd): route Anthropic Messages and Responses API through dual dispatch

### PR DESCRIPTION
**Pull request by pallasathena92**
### Problem

The HTTP PD router implements `/generate`, `/v1/chat/completions`, `/v1/completions`, and `/v1/rerank`, but `/v1/messages` (Anthropic Messages API) and `/v1/responses` (Responses API) fall through to the `RouterTrait` 501 default. The regular HTTP router already proxies both endpoints, so switching a deployment to PD mode silently loses two APIs. PD e2e coverage had the same gap: chat completions and gRPC-mode Responses only.

### Solution

Implement `route_messages` and `route_responses` on `PDRouter` following the exact shape of `route_chat`: derive the routing surface (cache-aware request text via `extract_text_for_routing`, rid key, stream flag from the request), build a `PDRequestContext` for the endpoint, and enter `execute_dual_dispatch`. No dispatch-layer changes were needed — PD bootstrap injection operates on serialized JSON (`inject_bootstrap_into_value`), `context.route` is used directly as the worker path, and `route_to_endpoint` already has metrics labels for both endpoints. Neither API has a batch parameter (`batch_size: None`) or PD logprob merging (`return_logprob: false`).

E2e: a new `test_pd_messages.py` drives the Anthropic SDK against the gateway over both PD wires — `pd_http` (SGLang dual dispatch, the path this PR adds) and `pd_grpc` (the mode-parameterized Messages pipeline, SGLang + vLLM) — and `test_pd_responses.py` gains a `pd_http` class for create and streaming. The HTTP Responses class deliberately skips storage semantics (retrieval, chaining): in proxy mode those belong to the engine, not the gateway.

## Changes

- `model_gateway/src/routers/http/pd_router.rs`: add `route_messages` and `route_responses` to the `RouterTrait` impl; import the two request types; add a route-level test proving both endpoints reach PD worker selection (503 on an empty fleet) instead of returning 501
- `e2e_test/router/test_pd_messages.py`: new — Messages API non-streaming + streaming over `pd_http` (SGLang) and `pd_grpc` (SGLang, vLLM)
- `e2e_test/router/test_pd_responses.py`: add `TestPDResponsesHttp` (`pd_http`, SGLang) covering create + streaming

## Test Plan

- New unit test `messages_and_responses_endpoints_dispatch_through_pd`: both endpoints on an empty-fleet PD router return `503 Service Unavailable` (PD selection reached) rather than `501 Not Implemented`
- `cargo test -p smg --lib`: 1797 passed, 0 failed; `cargo test -p smg --test routing_tests`: 120 passed
- `cargo clippy -p smg --all-targets -- -D warnings` clean, `cargo +nightly fmt` clean
- E2e: all 14 tests in the two touched files collect cleanly (`pytest --collect-only`); `ruff check` and `ruff format --check` clean. GPU execution rides the existing PD lanes: the `pd_grpc` Messages class runs on the vLLM PD lane; `pd_http` classes run on SGLang PD runs

<details>
<summary>Checklist</summary>

- [x] `cargo +nightly fmt` passes
- [x] `cargo clippy --all-targets -- -D warnings` passes for the touched crate
- [ ] (Optional) Documentation updated
- [ ] (Optional) Please join us on Slack [#sig-smg](https://slack.lightseek.org) to discuss, review, and merge PRs

</details>

### CODE REVIEW

File: model_gateway/src/routers/http/pd_router.rs [hunk 1]
```diff
@@ -1918,6 +1993,37 @@ mod tests {
...
+            "model": "m",
+            "input": "hi",
+        }))
+        .expect("valid responses request");
+        let response = router
+            .route_responses(None, &tenant, responses, UNKNOWN_MODEL_ID)
+            .await;
+        assert_ne!(response.status(), StatusCode::NOT_IMPLEMENTED);
+        assert_eq!(response.status(), StatusCode::SERVICE_UNAVAILABLE);
+    }
```
[the remark this answers is not in the document]
Comment (pallasathena92) [reply model_gateway/src/routers/http/pd_router.rs:2117]: Addressed in b39b443f. Added `messages_and_responses_dispatch_to_their_worker_routes`: recording loopback stubs are registered as prefill and decode workers, and the test asserts both legs of each dispatch hit the exact `/v1/messages` and `/v1/responses` paths with `bootstrap_host`/`bootstrap_port`/`bootstrap_room` injected into the forwarded JSON. The empty-fleet test is kept for the 503 error-mapping contract.

File: e2e_test/router/test_pd_responses.py [hunk 2]
```diff
@@ -30,6 +31,47 @@
...
+
+        events = list(resp)
+        created_events = [e for e in events if e.type == "response.created"]
+        assert len(created_events) > 0
+
+        delta_events = [e for e in events if e.type == "response.output_text.delta"]
+        assert len(delta_events) > 0
+
+        completed_events = [e for e in events if e.type == "response.completed"]
+        assert len(completed_events) == 1
```
[the remark this answers is not in the document]
Comment (pallasathena92) [reply e2e_test/router/test_pd_responses.py:79]: Addressed in b39b443f. Both Responses streaming tests (pd_http and pd_grpc classes) now concatenate the `response.output_text.delta` payloads and assert the combined text is non-empty, and assert `completed_events[0].response.status == "completed"`.
